presentation/matrix_view.py

```diff
@@ -214,30 +235,34 @@ def get_text(metric: MetricType, cell: MeshColumn) -> str:
         return f"<b>{cell.packet_loss_percent.value:.1f}%</b>"
 
     def make_hover_text(self, mesh: MeshResults) -> List[List[str]]:
-        text = []
+        matrix_hover_text: List[List[str]] = []
         for row in reversed(mesh.rows):
-            text_col = []
+            column_hover_text: List[str] = []
             for col in row.columns:
                 from_agent = mesh.agents.get_by_id(row.agent_id)
                 to_agent = mesh.agents.get_by_id(col.agent_id)
                 conn = mesh.connection(from_agent.id, to_agent.id)
-                latency_ms = conn.latency_millisec.value
-                jitter_ms = conn.jitter_millisec.value
-                loss = conn.packet_loss_percent.value
                 distance_unit = self._config.distance_unit
                 distance = calc_distance(from_agent.coords, to_agent.coords, distance_unit)
-                text_col.append(
+                cell_hover_text = (
                     f"{from_agent.alias} -> {to_agent.alias} <br>"
                     + f"Distance: {distance:.0f} {distance_unit.value}<br>"
-                    + f"Latency: {latency_ms:.2f} ms <br>"
-                    + f"Jitter: {jitter_ms:.2f} ms <br>"
-                    + f"Loss: {loss:.1f}%"
                 )
-            text.append(text_col)
+                if conn.is_no_data():
+                    cell_hover_text += "NO DATA"
+                else:
+                    latency_ms = conn.latency_millisec.value
+                    jitter_ms = conn.jitter_millisec.value
+                    loss = conn.packet_loss_percent.value
+                    cell_hover_text += (
+                        f"Latency: {latency_ms:.2f} ms <br>" + f"Jitter: {jitter_ms:.2f} ms <br>" + f"Loss: {loss:.1f}%"
+                    )
+                column_hover_text.append(cell_hover_text)
+            matrix_hover_text.append(column_hover_text)
         for i in range(len(mesh.rows)):
```

Comment:
Have you considered using `enumerate` in nested loops constructing the `matrix_hover_text`, so that you can handle cells on the diagonal right there without having to iterate over matrix again? Like this (starting at line 218):
```
for i, row in enumerate(reversed(mesh.rows)):
    ...
    for j, col in enumerate(row.columns):
        if i == j:
            column_hover_text.append("")
            continue
        ...
```

Reply:
This assumes the input data already contains such diagonal which you detect with "if i == j", but this is not the case. You need to add the diagonal manually

Reply:
You are right. However, my approach will work once we pre-create the matrix based on configured agents (as opposed to based on data available in "health" response.